misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestProcessor.java

```diff
@@ -147,30 +147,49 @@ private Map<Document, SyntaxTree> getTestSyntaxTreeMap(Module module) {
      * @return TestSuite
      */
     private TestSuite generateTestSuite(Module module, JarResolver jarResolver) {
-        PackageID packageID = module.descriptor().moduleTestCompilationId();
-        String testModuleName = packageID.isTestPkg ? packageID.name.value + Names.TEST_PACKAGE : packageID.name.value;
+        String testModuleName = getTestModuleName(module);
+        TestSuite testSuite = createTestSuite(module, testModuleName);
+
+        //skip if the cloud is set
+        if (jarResolver != null && module.project().buildOptions().cloud().isEmpty()) {
+            addTestExecutionDependencies(module, jarResolver, testSuite);
+        }
+
+        // TODO: Remove redundancy in addUtilityFunctions
```

Comment:
Let's mention this in the issue and keep track.

Reply:
This TODO was already there before the modification